modules/run-consul/run-consul

```diff
@@ -299,23 +299,33 @@ EOF
   fi
 
   local rpc_encryption_configuration=""
-  if [[ "$enable_rpc_encryption" == "true" && ! -z "$ca_path" && ! -z "$cert_file_path" && ! -z "$key_file_path" ]]; then
+  if [[ "$enable_rpc_encryption" == "true" && ! -z "$ca_path" ]]; then
     log_info "Creating RPC encryption configuration"
     rpc_encryption_configuration=$(cat <<EOF
 "verify_outgoing": true,
-"verify_incoming": true,
 "verify_incoming_https": false,
 "verify_server_hostname": true,
 "enable_agent_tls_for_checks": true,
-"ca_path": "$ca_path",
-"cert_file": "$cert_file_path",
-"key_file": "$key_file_path",
 "ports": {
   "https": 8501,
   "grpc": 8502
 },
+"ca_path": "$ca_path",
+EOF
+)
+    if [[ ! -z "$cert_file_path" && ! -z "$key_file_path" ]]; then
+        rpc_encryption_configuration+=$(cat <<EOF
+"verify_incoming": true,
+"cert_file": "$cert_file_path",
```

Comment:
I think this segment would benefit from having the `verify_incoming_https: false` definition.